src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversion.cs

```diff
@@ -914,7 +920,15 @@ public bool Equals(Conversion other)
         /// <returns>A hash code for the current <see cref="Conversion"/> object.</returns>
         public override int GetHashCode()
         {
-            return Hash.Combine(this.Method, (int)this.Kind);
+            int result = Hash.Combine(this.Method, (int)this.Kind);
+
+            var underlyingConversions = this.UnderlyingConversions;
+            if (!underlyingConversions.IsDefault)
+            {
+                result = Hash.Combine(result, Hash.CombineValues(underlyingConversions.SelectAsArray(c => c.GetHashCode())));
+            }
```

Comment:
`Hash.CombineValues` calls `GetHashCode()` on the `ImmutableArray<T>` and handles `IsDefault`.